cmdtests/sql_test.go

```diff
@@ -224,26 +222,27 @@ func (s *SQLTestSuite) TestIndexesWorkdirChange() {
 	// wait for gitbase to be ready
 	r = s.RunCommand("sql", "select 1")
 	require.NoError(r.Error, r.Combined())
-	// wait for gitbase to load index
-	time.Sleep(1 * time.Second)
 
 	s.testQueryWithIndex(require, "repos", true)
 }
 
 func (s *SQLTestSuite) testQueryWithIndex(require *require.Assertions, repo string, hasIndex bool) {
-	r := s.RunCommand("sql", "SHOW INDEX FROM repositories")
-	require.NoError(r.Error, r.Combined())
-
 	if hasIndex {
-		// parse result and check that correct index was built and it is visiable
-		indexLine := strings.Split(r.Stdout(), "\n")[3]
-		expected := `repositories.repository_id`
-		require.Contains(indexLine, expected)
-		visibleValue := strings.TrimSpace(strings.Split(indexLine, "|")[14])
+		var visibleValue string
+		// building/loading index takes some time, wait maximum 15s
+		for i := 0; i < 15; i++ {
```

Comment:
For the simple index we are creating in this testcase, 15s can be enough.
with bigger datasets (as could be needed in the future), it would not be enough.

Reply:
15s is a lot. I would prefer it to fail when we add such a big dataset. And increase timeout/reduce dataset depends on the case.